### PR TITLE
feat(data-stream): add fuel-core-nats V2

### PR DESCRIPTION
**Pull request by lostman**
`fuel-core-nats` is an extended version of `fuel-core`, which adds NATS publishing functionality.

See `crates/fuel-core-nats/README.md` for `fuel-core-nats` details.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -47,6 +47,7 @@ jobs:
         uses: dtolnay/rust-toolchain@stable
         with:
           toolchain: ${{ env.RUST_VERSION }}
+          targets: wasm32-unknown-unknown
```
Comment (pedronauck) [inline .github/workflows/ci.yaml:50]: why do we need wasm if we will not build for it?

File: .typos.toml [hunk 2]
```diff
@@ -1,2 +1,2 @@
 [default]
-extend-ignore-identifiers-re = ["TRO", "tro", "Tro"]
+extend-ignore-identifiers-re = ["TRO", "tro", "Tro", "16Uiu2HAm6pmJUedRFjennk4A8yWL6zCApHCuykzRRroqMjjxZ8o6", "16Uiu2HAm8dBwTRzqazCMqQDdR8thMa7BKiW4ep2B4DoQQp6Qhyfd"]
```
Comment (pedronauck) [inline .typos.toml:2]:
```suggestion
extend-ignore-identifiers-re = ["TRO", "tro", "Tro"]

[files]
extend-exclude = ["crates/**/README.md"]
```

File: Cargo.toml [hunk 3]
```diff
@@ -16,6 +16,10 @@ anyhow = "1.0"
 clap = { version = "4.5", features = ["derive"] }
 futures = "0.3"
 tokio = { version = "1.38", features = ["full"] }
+fuel-core-bin = { version = "0.27", features = ["env", "p2p", "relayer", "rocksdb"] }
```
Comment (pedronauck) [inline Cargo.toml:19]: any specific reason you are not using the latest `fuel-core`?

File: crates/fuel-core-nats/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+[package]
+name = "fuel-core-nats"
```
Comment (pedronauck) [inline crates/fuel-core-nats/Cargo.toml:2]: can you change for `fuel-data-stream` as we talked in the last sync?

**Review by pedronauck — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (pedronauck) [reply .github/workflows/ci.yaml:50]: ✅

File: crates/fuel-core-nats/Cargo.toml (hunk 4 above)
Comment (pedronauck) [reply crates/fuel-core-nats/Cargo.toml:2]: ok, let's go with this one as it is, then after the tech spec and more clarity about crates and etc we can bike-shedding about the name again

**Review by pedronauck — APPROVED** [review]

**Review by Jurshsmith — APPROVED** [review]

**Pull request merged**